File: backend/api/routes/admin/faq.py
from fastapi import APIRouter, Depends, status

from backend.api.dependencies import get_faq_service
from backend.api.schemas.faq import (
    FaqCreate,
    FaqResponse,
    FaqUpdate,
)
from backend.services.faq_service import FaqService


router = APIRouter(
    prefix="/faq",
    tags=["Admin — FAQ"],
)


@router.post(
    "",
    response_model=int,
    status_code=status.HTTP_201_CREATED,
)
async def create_faq(
    data: FaqCreate,
    service: FaqService = Depends(
        get_faq_service,
    ),
):
    return await service.create(
        question=data.question,
        answer=data.answer,
        position=data.position,
        is_hidden=data.is_hidden,
    )


@router.get(
    "/{faq_id}",
    response_model=FaqResponse,
)
async def get_faq(
    faq_id: int,
    service: FaqService = Depends(
        get_faq_service,
    ),
):
    return await service.get_by_id(
        faq_id,
    )


@router.get(
    "",
    response_model=list[FaqResponse],
)
async def get_faqs(
    is_hidden: bool | None = None,
    service: FaqService = Depends(
        get_faq_service,
    ),
):
    return await service.get_all(
        is_hidden=is_hidden,
    )


@router.put(
    "/{faq_id}",
)
async def update_faq(
    faq_id: int,
    data: FaqUpdate,
    service: FaqService = Depends(
        get_faq_service,
    ),
):
    await service.update(
        faq_id=faq_id,
        question=data.question,
        answer=data.answer,
        position=data.position,
        is_hidden=data.is_hidden,
    )

    return {
        "message": "FAQ обновлен."
    }


@router.delete(
    "/{faq_id}",
)
async def delete_faq(
    faq_id: int,
    service: FaqService = Depends(
        get_faq_service,
    ),
):
    await service.delete(
        faq_id,
    )

    return {
        "message": "FAQ удален."
    }

File: frontend/src/api/faq.ts
import { apiRequest } from './client'
import type {
  FaqCreate,
  FaqUpdate,
  FaqResponse,
} from '../types/faq'

export function getPublicFaq(): Promise<FaqResponse[]> {
  return apiRequest<FaqResponse[]>('/faq')
}

export function getPublicFaqItem(
  faqId: number,
): Promise<FaqResponse> {
  return apiRequest<FaqResponse>(
    `/faq/${faqId}`,
  )
}

export function getAdminFaq(
  isHidden?: boolean,
): Promise<FaqResponse[]> {
  const query = isHidden === undefined
    ? ''
    : `?is_hidden=${isHidden}`

  return apiRequest<FaqResponse[]>(
    `/admin/faq${query}`,
  )
}

export function getAdminFaqItem(
  faqId: number,
): Promise<FaqResponse> {
  return apiRequest<FaqResponse>(
    `/admin/faq/${faqId}`,
  )
}

export function createFaq(
  data: FaqCreate,
): Promise<number> {
  return apiRequest<number>('/admin/faq', {
    method: 'POST',
    body: JSON.stringify(data),
  })
}

export function updateFaq(
  faqId: number,
  data: FaqUpdate,
): Promise<{ message: string }> {
  return apiRequest<{ message: string }>(
    `/admin/faq/${faqId}`,
    {
      method: 'PUT',
      body: JSON.stringify(data),
    },
  )
}

export function deleteFaq(
  faqId: number,
): Promise<{ message: string }> {
  return apiRequest<{ message: string }>(
    `/admin/faq/${faqId}`,
    {
      method: 'DELETE',
    },
  )
}